zio-http/shared/src/main/scala/zio/http/internal/FormState.scala

```diff
@@ -25,69 +25,88 @@ private[http] sealed trait FormState
 
 private[http] object FormState {
 
-  final case class FormStateBuffer(
-    tree: Chunk[FormAST],
-    phase: FormState.Phase,
-    buffer: Chunk[Byte],
-    lastByte: Option[Byte],
+  final class FormStateBuffer(
+    private var lastByte: OptionalByte,
     boundary: Boundary,
-    dropContents: Boolean,
   ) extends FormState { self =>
 
-    def append(byte: Byte): FormState = {
+    private val tree0: ChunkBuilder[FormAST] = ChunkBuilder.make[FormAST]()
```

Comment:
I see you did for `FormState` what I suggested for `State`.
Did you try it for `State`? Did it fail?